hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractUpdateHandler.java

```diff
@@ -74,6 +107,124 @@ private boolean hasCryptoAdminKey(final ContractUpdateTransactionBody op) {
 
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
-        throw new UnsupportedOperationException("Not implemented");
+        final var txn = requireNonNull(context).body();
+        final var op = txn.contractUpdateInstanceOrThrow();
+        final var target = op.contractIDOrThrow();
+
+        final var accountStore = context.readableStore(ReadableAccountStore.class);
+
+        // validate update account exists
+        final var toBeUpdated = accountStore.getContractById(target);
+        validateTrue(toBeUpdated != null, INVALID_CONTRACT_ID);
+
+        final var changed = update(toBeUpdated, context, op);
+
+        context.serviceApi(TokenServiceApi.class).updateContract(changed);
```

Comment:
I believe we should contract records here ... after updating anything in accountStore using `TokenServiceAPI`

Reply:
Sounds good. I added the record but I'm not 100% sure what needs to be done.
I created a new `ContractUpdateRecordBuilder` class and for now, I set the `contractID`.
Do I need to record anything else? The updated contract(Account) probably?  I see that the ContractCreateHandler is recording `ContractFunctionResult` but I don't have it in my case. Maybe I should map it to that?

Reply:
Oh yeah, I don't think we need to set anything for ContractUpdate. So we don't need to do anything here and can remove lines 108, 109.  Sorry, did not realize 😄

Reply:
No problem 😄 Removed